### PR TITLE
Adds Prometheus metrics when in scheduled mode

### LINKED ISSUE #50: Enabling observability with metrics pushed
I've got some dashboards and alerts set up for my home lab. Now that I've got gickup running regularly, I'd like to be able to push some statistics to Influx from gickup. Adding this to gickup itself will add some complexity so I want to see what others think before I grab some [glögg](https://en.wikipedia.org/wiki/Gl%C3%B6gg) and have at it. 

I thought about Prometheus but I think the current architecture of gickup lends itself to push metrics rather than pull. TBH I've only used either Influx or Prom in recent history and they're setup in my home lab infra so I have a preference for those and only those.

### PR DESCRIPTION
**Pull request by colindean**
As submitted, this also pulls in InfluxDB2 libraries and sets up the stuff, but I don't have it doing anything with InfluxDB2 but constructing a client right now. By the time I finish this sprint of work, I'll probably _drop_ the InfluxDB2 implementation and come back to it later. 

My chief beef right now is that I want _one_ metrics API that can federate to both InfluxDB2 and Prometheus and whatever else and I've not found an existing library to do that. I'm going to end up implementing something for that if I keep InfluxDB2 and it might be... extra and not worthwhile.

I also did some refactoring in this PR while scoping out where in `local.go` I could add some metrics. That work might be worth its own separate PR...

Fixes #50

### CODE REVIEW

**Comment by colindean** [comment]
@cooperspencer What are your feelings here? What kind of additional counters, gauges, etc. would _you_ find useful?

I'll return to this code this week.

**Comment by cooperspencer** [comment]
Hi,
I will take a look at it tomorrow.

File: types/types.go [hunk 1]
```diff
@@ -31,6 +39,24 @@ type Conf struct {
...
+}
+
+type PrometheusConfig struct {
+	ListenAddr string `yaml:"listen_addr"`
+	Endpoint   string `yaml:"endpoint"`
+}
+
+type Metrics struct {
+	InfluxDb2  InfluxDb2Config  `yaml:"influxdb2"`
+	Prometheus PrometheusConfig `aml:"prometheus"`
```
Comment (cooperspencer) [inline types/types.go:59]: this should say 'yaml'

File: main.go [hunk 2]
```diff
@@ -135,7 +144,7 @@ func RunBackup(conf *types.Conf) {
 	repos = bitbucket.Get(conf)
 	Backup(repos, conf)
 
-	conf.HasValidCronSpec()
```
Comment (cooperspencer) [inline main.go:138]: I would actually leave it there because I think it is useful to know when the next run will be.

**Comment by cooperspencer** [comment]
overall it looks good, but it would be nice to have a metric of how long the backups overall took

File: main.go (hunk 2 above)
Comment (colindean) [reply main.go:139]: I refactored this in 1d7e22e so that it reads a little clearer and the log output isn't merely a side effect of an extra validation.

**Comment by colindean** [comment]
Added some more stats. I'm pretty happy with what's being output now. I'll build a dashboard in another PR. Let's get this merged before I do any additional refactoring.

**Pull request merged**